### PR TITLE
doc/install/manual-deployment: update osd creation steps

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/install/manual-deployment.rst [hunk 1]
```diff
@@ -348,116 +348,70 @@ Long Form
...
+     UUID=$(uuidgen)
 
-	ssh {new-osd-host}
-	sudo mkfs -t {fstype} /dev/{hdd}
-	sudo mount -o user_xattr /dev/{hdd} /var/lib/ceph/osd/{cluster-name}-{osd-number}
+#. Generate a cephx key for the OSD. ::
 
-	
-#. Initialize the OSD data directory. :: 
+     OSD_SECRET=$(ceph-authtool --gen-print-key)
```
Comment (alfredodeza) [inline doc/install/manual-deployment.rst]: there is an alternative to `--gen-print-key` right? or is this the only way of achieving this

File: doc/install/manual-deployment.rst [hunk 2]
```diff
@@ -348,116 +348,70 @@ Long Form
...
+     mkfs.xfs /dev/{DEV}
+     mount /dev/{DEV} /var/lib/ceph/osd/ceph-$ID
 
+#. Write the secret to the OSD keyring file.::
 
-#. Add the OSD to the CRUSH map so that it can begin receiving data. You may
-   also decompile the CRUSH map, add the OSD to the device list, add the host as a
-   bucket (if it's not already in the CRUSH map), add the device as an item in the
-   host, assign it a weight, recompile it and set it. ::
+     printf "[osd.$ID]\n\tkey = $OSD_SECRET\n" > /var/lib/ceph/osd/ceph-$ID/keyring
```
Comment (alfredodeza) [inline doc/install/manual-deployment.rst]: The way we are doing this is with `ceph-authtool`, like:

    ceph-authtool /var/lib/ceph/osd/ceph-$ID/keyring --create-keyring --name osd.$ID --add-key $OSD_SECRET

Is that wrong? are we just trying to exemplify the most manual way here?

File: doc/install/manual-deployment.rst [hunk 3]
```diff
@@ -348,116 +348,70 @@ Long Form
...
-	ceph [--cluster {cluster-name}] osd crush add {id-or-name} {weight} [{bucket-type}={bucket-name} ...]
+#. Initialize the OSD data directory. ::
 
-   For example::
+     ceph-osd -i $ID --mkfs --osd-uuid $UUID
 
-	ceph osd crush add osd.0 1.0 host=node1
+#. Fix ownership.::
 
+     chown -R ceph.ceph /var/lib/ceph/osd/ceph-$ID
```
Comment (alfredodeza) [inline doc/install/manual-deployment.rst]: I think this is `ceph:ceph`

File: doc/install/manual-deployment.rst (hunk 1 above)
Comment (liewegas) [inline doc/install/manual-deployment.rst]: Several different ways to get the key value.. this is just the easiest

File: doc/install/manual-deployment.rst (hunk 2 above)
Comment (liewegas) [inline doc/install/manual-deployment.rst]: Yeah, that works too.  Any preference?

File: doc/install/manual-deployment.rst (hunk 3 above)
Comment (liewegas) [inline doc/install/manual-deployment.rst]: fixed!

File: doc/install/manual-deployment.rst (hunk 1 above)
Comment (liewegas) [inline doc/install/manual-deployment.rst]: There's a way to write the keyring *and* generate the key and print to stdout in one step, but we don't know the osd id yet at this stage, so --gen-print-key is what we want.

File: doc/install/manual-deployment.rst (hunk 2 above)
Comment (liewegas) [inline doc/install/manual-deployment.rst]: went with your suggestion.. cleaner i think

**Pull request merged**